### ISSUE TITLE
ethdagsize.json conflict

### ISSUE THREAD

**Issue by 0xSoliski** [RainbowMiner/RainbowMiner#2368]
Scenario:
- GPU with 4GB ram
- Mining on ZergPoolCoins
- Algorithm: Ethash or KawPow

Expected behavior:
RBM picks the most profitable coin that needs <4GB VRAM. In order to check if the coin needs <4GB, RBM looks at the ethdagsizes.json to determine eligibility. If the coin is not on the list, go to the next profitable coin until you find one that is on the list in order to be able to verify that it needs <4GB.

Actual behavior:
RBM picks the most profitable coin in that algorithm. If it's not on the list, it disables the algorithm altogether. In a recent example, GPN on KaPOW was the most profitable and can fit in <4GB but because it was not on ethdagsize.json the KaPOW algorithm was excluded instead of falling back to CLORE or NEOX.

![image](https://github.com/RainbowMiner/RainbowMiner/assets/22576563/e84511ac-fb15-4c50-ae49-d733e165fe88)
In the above scenario, KawPOW will be disabled for 4GB cards as well because RVN is the most profitable and cannot fit in <4GB.
Temporary solution: Disable RVN in config.json and it will start mining CLORE

TLDR; algos are disabled if the most profitable coin cannot fit in the available VRAM (or RBM things they cannot fit because they are not in ethdagsize.json) instead of falling back to the next available one.

**Comment by RainbowMiner** [comment]
Good finding! This can be easily solved - I'll change the Get-EthdagSizes function, so that it returns 1Gb for all coins, that aren't in the ethdagsizes.json database file.

**Comment by sfalas01** [comment]
> Good finding! This can be easily solved - I'll change the Get-EthdagSizes function, so that it returns 1Gb for all coins, that aren't in the ethdagsizes.json database file.

This will fix one part of the issue, yes. This will not fix the issue where a coin does exist in the ethdagsizes.json but does not fit into the VRAM, mainly for KaPOW, because it does not have variants as Ethash does. In the screenshot example, if I don't add RVN into the excluded coins, KaPOW is not an option to mine even though CLORE is mineable.
The fix for this would need to be:
1) Either KaPOW variants, like Ethash, in order to differentiate between them, or
2) If the most profitable coin to mine does not fit into VRAM, then it skips it and moves on to the next profitable and so on, until it finds one that fits

**Comment by RainbowMiner** [comment]
Understood. One possible solution would be to differentiate the algorithm by coin. Like KawPowRVN and KawPowCLORE. Unfortunately, we would need to benchmark each algo/coin on it's own, then. Another problem would be to handle dual mining configurations in the miner modules.
There are better solutions, I'll have to think that through completely.

**Comment by sfalas01** [comment]
Well, in general, the process now seems to be:
1) Find the most profitable coin per algo (for the ZergPoolCoins example here)
2) Check if it fits in the VRAM
3) Add it to the list of profitability

Instead what I would do (again talking out of thin air here, just an idea that maybe breaks a lot more things than it solves :P)
1) Create a list with eligible coins per GPU (the ones that fit)
2) Find the most profitable coin per algo (for the ZergPoolCoins example here)
3) Add it to the list of profitability

So essentially, the screening of coins would happen before even profitability is considered.

**Comment by UselessGuru** [comment]
> There are better solutions, I'll have to think that through completely.

My approach:
If the pool provides the currency (or coin for that matter) then I get the DAG size for that coin.
If the pool does not provide the currency (e.g. NiceHash) then use the largest DAG size for that algo.
https://github.com/Minerx117/NemosMiner/blob/9a397d1bc919293816d763b041d6e89754a1462b/Includes/Core.ps1#L379


For that reason I keep a list of DAG sizes for each known DAG algoritchm as well as a list of DAG sizes for all known coins:
https://github.com/Minerx117/NemosMiner/blob/master/Data/DagData.json

**Comment by UselessGuru** [comment]
Please check your PM in discord.

**Comment by RainbowMiner** [comment]
The scenario here is ethdagsizes.json vs. ZergPoolCoins on a 4GB card. ZergPool was removed from RainbowMiner on 2025-09-27 (b4f6f448). Closing as obsolete for the current master (v5.0.2.1) - please open a new issue if this still occurs. Thanks for the report!
